### ISSUE TITLE
ASNetworkImageNode should complain about having its image property set directly.

### ISSUE THREAD

**Issue by jeffdav** [facebook/AsyncDisplayKit#2275]
`ASNetworkImageNode` inherits from `ASImageNode`.  If you set the inherited `image` property directly, e.g.:

```
private let imageNode = ASNetworkImageNode()

init(...) {
  imageNode.image = UIImage(contentsOfFile: "/example/image.png")
}
```

... this works.  

If, however, it is a `ASHierarchyStateRangeManaged` node (in my case, a child of an `ASCellNode`), once its interface state is set to not have `ASInterfaceStateFetchData`, the image is cleared.  When the cell is displayed again, the image is not reloaded since the `fetchData` implementation doesn't have an URL.

One solution is to assert in the `ASNetworkImageNode` subclass if you try to set the image direcly and the hierarchy state has the managed flag.

**Comment by garrettmoon** [comment]
You're right and an assert is a good solution.

**Comment by hannahmbanana** [comment]
@jeffdav: thanks for catching this!

**Comment by garrettmoon** [comment]
@jeffdav we're tracking this internally, but please feel free to put up a pull request!

**Comment by hannahmbanana** [comment]
@garrettmoon - was this resolved by either #2782 or #2708?

**Comment by hannahmbanana** [comment]
@jeffdav - This should now behave reliably after PR #2782 and #2708. Please let us know if you see any incorrect behavior. Thanks!
